pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java

```diff
@@ -58,12 +61,13 @@
   public static final String ERROR = "ERROR";
   public static final String CONSUMING = "CONSUMING";
   private ScheduledExecutorService _executorService;
-  ControllerMetrics _metricsRegistry;
+  private ControllerMetrics _metricsRegistry;
   private ControllerConf _config;
   private final PinotHelixResourceManager _pinotHelixResourceManager;
   private final HelixAdmin _helixAdmin;
   private final long _segmentStatusIntervalSeconds;
   private final int _waitForPushTimeSeconds;
+  private TableSizeReader _tableSizeReader;
```

Comment:
Do we need a setter for this? If not, make it final

Reply:
We do need a setter for this, otherwise there's no way for us to mock the reader's behavior in the test.